tools/tsp-client/README.md

```diff
@@ -2,45 +2,65 @@
 
```

Comment:
Should we update the binary name and the header above? It feels confusing to have a mismatch between the binary command name and the package name.

Not that I think many will thank you for making them type out `typespec-client-generator-cli` in the terminal ;)

Reply:
Yeah I know it's not perfect :/ but usage wise I think users will be more thankful to have a shorter alias to type out.

Reply:
Plus this kind of keeps us in sync with the `tsp` cli, by making this `tsp-client` so hopefully the change isnt too jarring for folks who are moving from one to the other